resources/axon.rb

```diff
@@ -98,82 +99,132 @@
     not_if { new_resource.tags.empty? }
   end
 
-  # Set local cache target for the installers
-  local_installer = ::Chef::Config['file_cache_path'] + '/te_agent' + ext
+  agent_source_is_url = new_resource.installer.start_with?('http')
+  agent_source_is_tgz = new_resource.installer.end_with?('tgz')
+  agent_source_is_zip = new_resource.installer.end_with?('zip')
 
   # Set the correct header for the agent installer
-  agent_source = if new_resource.installer.start_with?('http')
-                   new_resource.installer
-                 else
-                   'file:///' + new_resource.installer
-                 end
-
-  # Download installer
-  remote_file local_installer do
-    source agent_source
-    mode '744' unless node['platform'] == 'windows'
+  agent_source = agent_source_is_url ? new_resource.installer : 'file:///' + new_resource.installer
+  folder = ::File.basename(agent_source, '.*')
+  axon_chef_cache = "#{::Chef::Config['file_cache_path']}/#{folder}"
+
+  # Set local cache target for the installers
+  node.run_state['local_installer'] = "#{::Chef::Config['file_cache_path']}/te_axon#{ext}"
+
+  # Download and unzip the installer
+  if agent_source_is_tgz
+    tar_extract agent_source do
+      target_dir ::Chef::Config['file_cache_path']
+      action :extract
+      # Use the license file for idempotency
+      creates "#{axon_chef_cache}/license.html"
+    end
+  elsif agent_source.end_with?('zip')
```

Comment:
you created the `agent_source_is_zip` variable for this, no?

Reply:
I did! Thanks, will update that.